activerecord/lib/active_record/associations/association.rb

```diff
@@ -77,13 +101,13 @@ def reload(force = false)
         self unless target.nil?
       end
 
-      # Has the \target been already \loaded?
+      # Has the +target+ been already loaded?
       def loaded?
```

Comment:
Should this be nodoc as well?
```suggestion
      def loaded? # :nodoc:
```

Reply:
Same thing applies to `stale_target?`, `scope`, `strict_loading`, etc.

Reply:
The idea here was to keep a small set of methods public which I felt were useful, and mostly the read-only ones except strict_loading.

That's why like `owner=` is not public.

Reply:
I think `nodoc`ing methods that we aren't certain people are using,  especially writers, is a good idea.

Here's the current list now:

- [collection?](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-collection-3F)
- [create](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-create)
- [create!](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-create-21)
- [extensions](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-extensions)
- [klass](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-klass)
- [load_target](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-load_target)
- [loaded?](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-loaded-3F)
- [reload](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-reload)
- [scope](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-scope)
- [stale_target?](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-stale_target-3F)
- [target](https://zzak-public-association-api.rails-docs-preview.pages.dev/api/classes/ActiveRecord/Associations/Association.html#method-i-target)

Reply:
I would reduce this to the methods we know we want to make public for now.
We can always add methods, but removing methods requires a deprecation cycle.